pegasus/sites.v3/code.org/public/v3/edit-csp-table/splat.haml

```diff
@@ -151,6 +151,21 @@ angular: recordsApp
       column.id = column.name;
       column.editing = false;
     }
+    $scope.coerceColumn = function (column, type) {
+      $http({
+        method:'POST',
+        url: '/v3/coerce-shared-tables/#{channel_id}/#{table_name}',
+        params: {column_name: column.id, type: type}
+      }).then(function (response) {
+        var conversionError = '';
+        if (response.data !== 'true') {
+          // reload the page with conversionError set, allowing us to show an error string
+          conversionError = '?conversionError=Not all values in column ' + column.id +
+            ' could be converted to type ' + type + '.';
+        }
+        location.assign(location.pathname + conversionError);
+      });
```

Comment:
Oh, I see now that you're reloading to fetch the new data. You should be able to do something like `$scope.records = recordsService.query()` to reload the data from the server.